pages/directory/index.tsx

```diff
@@ -1,13 +1,72 @@
-import { Box, Container, List, ListItem, Typography } from '@mui/material';
+import { Box, Container, CircularProgress, List, ListItem, Typography } from '@mui/material';
+import axios from 'axios';
 import { NextPage } from 'next';
 import React from 'react';
 
 import createAxiosInstance from 'client/lib/axios';
 
 //TODO: Type users once sequelize-typescript is added
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const Directory: NextPage<{ users: any[]; error: string }> = props => {
-  const { users, error } = props;
+const Directory: NextPage<{ users: any[]; totalPages: number; error: string }> = props => {
+  const { users, totalPages, error } = props;
+  const lastCardRef = React.useRef(null);
+  const [userData, setUserData] = React.useState(users);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [totalPageNum, setTotalPageNum] = React.useState(totalPages);
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  async function fetchUsers(page: number) {
+    try {
+      const response = await axios.get(`/api/users?page=${page}`);
+      return {
+        page: response.data.page,
+        total: response.data.totalPages,
+        users: response.data.users,
+      };
+    } catch (err) {
+      console.error(err);
```

Comment:
We should avoid using `console` on the client side as it will fail without notifying the user... see linked comments on #87 and the approach they took to address it.

https://github.com/timmyichen/dev-directory/pull/87#discussion_r1185713660
https://github.com/timmyichen/dev-directory/pull/87#discussion_r1186645305